src/glados/db.py

```diff
@@ -0,0 +1,290 @@
+        ).update(self.__sql_values__)
+
+
+class DataStore:
+    def __init__(
+        self,
+        host: str,
+        username: str,
+        password: str,
+        port: int = 5432,
+        database: str = "glados",
+    ):
+        self.host = host
+        self.port = port
+        self.username = username
+        self.password = password
+        self.database = database
+
+        self.db = create_engine(
+            f"postgres://{self.username}:{self.password}@{self.host}:{self.port}/"
+            f"{self.database}"
+        )
+
+    def create_session(self) -> Session:
+        """Generate a new session with the existing connection.
+
+        Returns
+        -------
+        Session:
+            the session that was generated
+        """
+        session = sessionmaker(self.db)  # type: sessionmaker
+        return session()
+
+    def table_exists(self, table=TABLE_INTERACTIONS) -> bool:
+        """Check to see if the GLaDOS table is found in postgres.
+
+        Returns
+        -------
+        bool:
+            True if the table is found. False if it is not found.
+
+        """
+        return table in self.db.table_names()
+
+    def drop_table(self, table=TABLE_INTERACTIONS, force=False):
+        """Drop the GLaDOS table so that it can be re-created.
+
+        Returns
+        -------
+        bool:
+            was the drop table successful.
+        """
+        Table(table, Metadata).drop(self.db, checkfirst=not force)
+
+    def create_table(self, tables: Optional[List[str]] = None, force=False):
+        """Create the table.
+
+        If you set force to True then it will drop the existing tables and
+        then recreate them. ALL DATA WILL BE LOST
+
+        Parameters
+        ----------
+        tables : Optional[List[str]]
+            only take action on these tables. If None, then take action on all tables
+        force : bool
+            drop existing tables and rebuild. (default: False)
+
+        Returns
+        -------
+
+        """
+        if force:
+            if tables:
+                for table in tables:
+                    self.drop_table(table)
+            else:
+                Base.metadata.drop_all(self.db, checkfirst=False)
+
+        if tables:
+            for table in tables:
+                Table(table, Metadata).create(self.db)
+        else:
+            Metadata.create_all(self.db)
+
+    def find_by_id(self, interaction_id: str, session: Session) -> DataStoreInteraction:
+        """Find an interaction by interaction_id.
+
+        Parameters
+        ----------
+        interaction_id : str
+            interaction ID to find
+        session : Session
+            session to be used
+
+        Returns
+        -------
+        DataStore :
+            The interaction object
+
+        """
+        result = session.query(DataStoreInteraction).get(interaction_id)
+
+        logging.debug(result.interaction_id)
+        return result.object
```

Comment:
The `result` is already an object... what's `.object` here?

Reply:
Hahaha debugging code